src/sensr_client.cpp

```diff
@@ -26,45 +26,43 @@ namespace sensr
   }
 
   void Client::Reconnect() {
-    if (!is_reconnecting_) {
+    if (is_reconnecting_) {
       StopReconnecting();
```

Comment:
I'm concerned that it keeps creating and joining a thread frequently. It would be great if it is an asynchronous tail-recursion. Here's a brief pseudocode:
```cpp
void Client::Reconnect() {
  if (!IsConnected() && !is_reconnecting_) {
    is_reconnecting_ = true;
    connect_async();
  }
  io_context_.poll();
}

void Client::connect_async() {
  if (connected) {
    // handle connection
  } else {
    connect_async();
  }
}
```

Reply:
got it. Let me apply your advice.